cncf-youtube-channel-summarizer/transcript_summarizer.py

```diff
@@ -0,0 +1,120 @@
+
+from langchain.text_splitter import RecursiveCharacterTextSplitter
+from ibm_watson_machine_learning.foundation_models import Model
+from langchain.chains.summarize import load_summarize_chain
+import json
+from langchain import PromptTemplate
+import pandas as pd
+import os
+from tenacity import retry, stop_after_attempt, wait_fixed
+from logger import setup_logger
+
+logger = setup_logger("TranscriptSummarizer")
+class TranscriptSummarizer():
+    def __init__(self, model_id, summary_param, keywords_param, transcript_path):
+        self.APIKEY = os.environ['WATSONX_KEY']
+        self.project_id = os.environ['WATSONX_PROJECT_ID']
+        self.model_id = model_id
+        self.summary_param = summary_param
+        self.keywords_param = keywords_param
+        f = open(transcript_path)
+        self.videos_dict = json.load(f)
+        self.llm_summary, self.llm_keywords = self.create_models()
+
+    def create_models(self):
+        llm_summary = Model(
+            model_id=self.model_id,
+            credentials={
+                "apikey": self.APIKEY,
+                "url": "https://us-south.ml.cloud.ibm.com"
+            },
+            project_id=self.project_id,
+            params={
```

Comment:
maybe you can make the `params`  and other reusable options at the global level instead of declaring them separately.